os/arch/arm/src/common/up_initialize.c

```diff
@@ -131,26 +131,16 @@ static void up_calibratedelay(void)
 #if defined(CONFIG_STACK_COLORATION) && CONFIG_ARCH_INTERRUPTSTACK > 3
```

Comment:
Hi, requesting you to consider re-phrasing the commit description a little: -
```
irq stack sets INTSTACK_COLOR when initialized, but while checking, STACK_COLOR is used in up_assert().
The setting logic and check logic must be consistent.

This commit changes irq_stack color to same STACK_COLOR during initialization.
Since we use stack color, we should check CONFIG_STACK_COLORATION.
This commit adds to check CONFIG_STACK_COLORATION in up_color_nestirqstack().
```